src/App.tsx

```diff
@@ -1,60 +1,187 @@
+                  userSelected={userSelected}
+                  handleUserPosts={fetchUserPosts}
+                  setShowSideBar={setIsSidebarVisible}
+                />
               </div>
 
-              <PostsList />
+              <div className="block" data-cy="MainContent">
+                {!userSelected && (
+                  <p data-cy="NoSelectedUser">No user selected</p>
+                )}
+
+                {showPostLoader && <Loader />}
+
+                {hasPostError && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    Something went wrong!
+                  </div>
+                )}
+
+                {userPosts.length > 0 && !showPostLoader && (
+                  <PostsList
+                    userPosts={userPosts}
+                    handleShowComment={fetchPostComments}
+                    showSideBar={isSidebarVisible}
+                    setShowSideBar={setIsSidebarVisible}
+                    setShowCommentForm={setIsCommentFormVisible}
+                  />
+                )}
+                {userSelected &&
+                  userPosts.length === 0 &&
+                  !showPostLoader &&
+                  !hasPostError && (
+                    <div
+                      className="notification is-warning"
+                      data-cy="NoPostsYet"
+                    >
+                      No posts yet
+                    </div>
+                  )}
+              </div>
             </div>
           </div>
-        </div>
 
-        <div
-          data-cy="Sidebar"
-          className={classNames(
-            'tile',
-            'is-parent',
-            'is-8-desktop',
-            'Sidebar',
-            'Sidebar--open',
-          )}
-        >
-          <div className="tile is-child box is-success ">
-            <PostDetails />
+          <div
+            data-cy="Sidebar"
+            className={classNames(
+              'tile',
+              'is-parent',
+              'is-8-desktop',
+              'Sidebar',
+              { 'Sidebar--open': isSidebarVisible },
+            )}
+          >
+            <div className="tile is-child box is-success ">
+              <PostDetails
+                isCommentLoading={showCommentLoader}
+                postComments={postComments}
+                selectedPost={selectedPost}
+                errorsComment={hasCommentError}
+                onDeleteComment={removeComment}
+                onAddComment={createComment}
+                isButtonAddLoading={isAddingComment}
+                onShowCommentFormChange={setIsCommentFormVisible}
+                isCommentFormVisible={isCommentFormVisible}
+                userSelected={userSelected}
+              />
+            </div>
           </div>
         </div>
       </div>
-    </div>
-  </main>
-);
+    </main>
+  );
```

Comment:
The render method is quite long and contains a lot of logic. It would be better to split it into smaller, well-named components to make the code more readable and maintainable.